exotica_python/src/pyexotica/tools.py

```diff
@@ -27,6 +29,29 @@ def check_trajectory_continuous_time(scene, trajectory):
     print("Continuous-time collision verification took", end_time - start_time)
     return all_good
 
+
+def check_whether_trajectory_is_collision_free_by_subsampling(scene, trajectory, num_subsamples=10, debug=False):
+    '''
+    num_subsamples specifies how many steps are checked between two configurations.
+    Returns True if trajectory is collision-free, and False otherwise.
+
+    TODO: Support setting time for Scene update.
+    '''
+    start_time = time()
+    trajectory_length = trajectory.shape[0]
+    for t in range(1, trajectory_length):
+        q_t_1 = trajectory[t-1, :]
+        q_t_2 = trajectory[t, :]
+        q_t_interpolation = np.linspace(q_t_1, q_t_2, num_subsamples)
+        for i in range(num_subsamples):
+            scene.update(q_t_interpolation[i, :])
+            if not scene.is_state_valid(True):
+                return False
+    end_time = time()
+    debug and print("Trajectory transition collision check took", end_time - start_time)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expression "(debug) and (print('Trajectory transition collision check took', (end_time) - (start_time)))" is assigned to nothing](https://app.codacy.com/gh/ipab-slmc/exotica/pullRequest?prid=5476070)